static/app/views/releases/detail/index.tsx

```diff
@@ -54,101 +51,63 @@ type ReleaseContextType = {
-} & DeprecatedAsyncView['state'];
-
-class ReleasesDetail extends DeprecatedAsyncView<Props, State> {
-  shouldReload = true;
-
-  getTitle() {
-    const {params, organization, selection} = this.props;
-    const {release} = this.state;
-
-    // The release details page will always have only one project selected
-    const project = release?.projects.find(p => p.id === selection.projects[0]);
-
-    return routeTitleGen(
-      t('Release %s', formatVersion(params.release)),
-      organization.slug,
-      false,
-      project?.slug
-    );
-  }
-
-  getDefaultState() {
-    return {
-      ...super.getDefaultState(),
-      deploys: [],
-      sessions: null,
-    };
-  }
-
-  componentDidUpdate(prevProps: Props, prevState: State) {
-    const {organization, params, location} = this.props;
-
-    if (
-      prevProps.params.release !== params.release ||
-      prevProps.organization.slug !== organization.slug ||
-      !isEqual(
-        this.pickLocationQuery(prevProps.location),
-        this.pickLocationQuery(location)
-      )
-    ) {
-      super.componentDidUpdate(prevProps, prevState);
-    }
-  }
-
-  getEndpoints(): ReturnType<DeprecatedAsyncComponent['getEndpoints']> {
-    const {organization, location, params, releaseMeta} = this.props;
-
-    const basePath = `/organizations/${organization.slug}/releases/${encodeURIComponent(
-      params.release
-    )}/`;
+function pickLocationQuery(location: Location) {
+  return pick(location.query, [
+    ...Object.values(URL_PARAM),
+    ...Object.values(PAGE_URL_PARAM),
+  ]);
+}
 
-    const endpoints: ReturnType<DeprecatedAsyncView['getEndpoints']> = [
-      [
-        'release',
-        basePath,
-        {
-          query: {
-            adoptionStages: 1,
-            ...normalizeDateTimeParams(this.pickLocationQuery(location)),
-          },
+function ReleasesDetail({
+  children,
+  releaseMeta,
+}: {
+  children: React.ReactNode;
+  releaseMeta: ReleaseMeta;
+}) {
+  const params = useParams<{release: string}>();
+  const organization = useOrganization();
+  const {selection} = usePageFilters();
+  const location = useLocation();
+  const releasePath = `/organizations/${organization.slug}/releases/${encodeURIComponent(
+    params.release
+  )}/`;
+
+  const {
+    data: release,
+    refetch: refetchRelease,
+    isPending: IsReleasePending,
+    error: releaseError,
+  } = useApiQuery<ReleaseWithHealth>(
+    [
+      releasePath,
+      {
+        query: {
+          adoptionStages: 1,
+          ...normalizeDateTimeParams(pickLocationQuery(location)),
         },
-      ],
-    ];
+      },
+    ],
+    {staleTime: Infinity}
+  );
 
-    if (releaseMeta.deployCount > 0) {
-      endpoints.push([
-        'deploys',
-        `${basePath}deploys/`,
-        {
-          query: {
-            project: location.query.project,
-          },
-        },
-      ]);
-    }
+  const {
+    data: deploys = [],
+    refetch: refetchDeploys,
+    isPending: IsDeploysPending,
```

Comment:
casing here seems a bit off and isPending will never be true while disabled

Reply:
I feel like I keep forgetting that, thank you! fixed